Source/WebCore/page/scrolling/ScrollAnchoringController.cpp

```diff
@@ -50,35 +53,138 @@ ScrollOffset ScrollAnchoringController::computeOffset(RenderBox& candidate)
     return ScrollOffset(FloatPoint(candidate.absoluteBoundingBoxRect().location() - m_frameView.layoutViewportRect().location()));
 }
 
-void ScrollAnchoringController::chooseAnchorElement()
+static Element* anchorElementForPriorityCandidate(Element* element)
 {
-    // TODO: fully implement anchor node selection algorithm in spec
-    // (specifically excluded subtrees and priority nodes)
-    // https://drafts.csswg.org/css-scroll-anchoring/#anchor-node-selection
+    while (element) {
+        if (auto renderer = element->renderBox()) {
+            if (!renderer->isAnonymousBlock() && (!renderer->isInline() || renderer->isAtomicInlineLevelBox()))
+                return element;
+        }
+        element = element->parentElement();
+    }
+    return nullptr;
+}
 
-    auto* document = m_frameView.frame().document();
-    if (!document)
-        return;
+bool ScrollAnchoringController::examinePriorityCandidate(Element* element)
+{
+    if (auto* candidateElement = anchorElementForPriorityCandidate(element)) {
+        LOG_WITH_STREAM(ScrollAnchoring, stream << "ScrollAnchoringController::examinePriorityCandidate()");
+        auto* iterElement = candidateElement;
+
+        // TODO: iterate up to owning scroller
+        while (iterElement) {
+            LOG_WITH_STREAM(ScrollAnchoring, stream << "ScrollAnchoringController::examinePriorityCandidate() for element: " << *iterElement << " element scrolling id: " << ((element->renderBox() && element->renderBox()->layer() && element->renderBox()->layer()->scrollableArea()) ? (element->renderBox()->layer()->scrollableArea()->scrollingNodeID()) : 0) << " current Frame: " << m_frameView << "frame: scrolling nodeID: " << m_frameView.scrollingNodeID());
 
-    Element* candidateAnchor = nullptr;
+            if (auto renderer = element->renderBox()) {
```

Comment:
Is there a reason to use `renderBox()` and not `renderer()`?

Reply:
Fixed.